webapp/src/app/services/phaser/helper.ts

```diff
@@ -115,15 +122,23 @@ export class Helper {
 			return true;
 		}
 		
-		// TODO: save promise to avoid loading the same texture simultaneously
+		if (this.cache[key]) {
+			return this.cache[key];
+		}
+		const promise = this.loadTextureInternal(key, scene);
+		this.cache[key] = promise;
+		return promise;
```

Comment:
I completely missed this code. This doesn't work because the cache has to be dependant on the scene that is passed in as an argument.